Squash TypeScript Warnings

As a developer of VZCode, I want to have confidence that everything in the codebase is properly typed.

Current behavior: running `npm run tsc` results in one warning.

```
vzcode$ npm run tsc

> vzcode@0.20.0 tsc
> tsc --noEmit

src/client/Sidebar/FileListing.tsx:43:6 - error TS2741: Property 'name' is missing in type '{ children: string; handleClick: () => void; handleDeleteClick: (event: MouseEvent<Element, MouseEvent>) => void; handleRenameClick: (newName: string) => void; isDir: boolean; }' but required in type '{ name: any; children: any; handleClick: any; handleRenameClick: any; handleDeleteClick: any; isDir: any; }'.

43     <Item
        ~~~~


Found 1 error in src/client/Sidebar/FileListing.tsx:43

```

Desired behavior: running `npm run tsc` results in no errors.

--- a/src/client/Sidebar/FileListing.tsx
+++ b/src/client/Sidebar/FileListing.tsx
@@ -45,6 +45,7 @@ export const FileListing = ({
       handleDeleteClick={handleDeleteClick}
       handleRenameClick={handleRenameClick}
       isDir={false}
+      name = {name}
     >
       {name}
     </Item>

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -2,7 +2,8 @@
   "declaration": true,
   "compilerOptions": {
     "jsx": "react-jsx",
-    "skipLibCheck": true
+    "skipLibCheck": true,
+    "esModuleInterop": true
   },
   "include": ["src/**/*"],
   "//": "TODO - remove this when we have this change: https://github.com/vitest-dev/vitest/issues/3622",

